core/contractsdk/go/example/award_manage/main.go

```diff
@@ -0,0 +1,253 @@
+package main
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/xuperchain/xuperchain/core/contractsdk/go/code"
+	"github.com/xuperchain/xuperchain/core/contractsdk/go/driver"
+	"github.com/xuperchain/xuperchain/core/contractsdk/go/utils"
+)
+
+const (
+	BALANCEPRE   = "balanceOf_"
+	ALLOWANCEPRE = "allowanceOf_"
+	MASTERPRE    = "admin"
+	TOTAL_SUPPLY = "TotalSupply"
+)
+
+type awardManage struct{}
+
+func (am *awardManage) Initialize(ctx code.Context) code.Response {
+	args := struct {
+		TotalSupply *big.Int `json:"totalSupply" gt:"0"`
+	}{}
+	if err := utils.Validate(ctx.Args(), &args); err != nil {
```

Comment:
Could the `utils.Validate` solve a number < 0?

Reply:
Yeah, the version @ commit hash  70f6c7f3 validates number range ,  implementing  using reflect and handwirtten code. 
What is more, I have a better Implementation using a more robust and expressive comminity library [validator](https://github.com/go-playground/validator) now.